plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/navigator/internal/BuiltInProjectNode.java

```diff
@@ -65,7 +65,14 @@ public String getText() {
 				.transform(p -> p.getLocation())
 				.transform(uri -> uri.toFileString())
 				.transform(uri -> new File(uri))
-				.transform(file -> new ResourceNode(this, file))
+				.transform(file -> {
+					try {
+						return new ResourceNode(this, file);
```

Comment:
make constructor private, expose factory method, move checks from the constructor to the factory method, instead of throwing exception return null

then filter null here, remove try-catch